xpti/include/xpti_trace_framework.hpp

```diff
@@ -0,0 +1,311 @@
+//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+#pragma once
+#include <sstream>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "xpti_data_types.h"
+#include "xpti_trace_framework.h"
+
+#if defined(_WIN32) || defined(_WIN64)
+#include <string>
+#include <strsafe.h>
+#include <windows.h>
+typedef HINSTANCE xpti_plugin_handle_t;
+typedef FARPROC xpti_plugin_function_t;
+#define XPTI_PLUGIN_STRING "*.dll"
+#define XPTI_PATH_SEPARATOR "\\"
+// Windows does not have PATH_MAX defined or is inconsistent; Documentation
+// suggests that 32767 is the max string length of environment variables on
+// Windows
+constexpr auto WIN_PATH_MAX = 32767;
+#else // Linux and MacOSX
+#include <cstdlib>
+#include <dirent.h>
+#include <dlfcn.h>
+#include <limits.h>
+#include <string.h>
```

Comment:
Perhaps now you need less `#include` since you removed a lot of code.

Reply:
Yes, I removed a couple of them.